fastlane/fastlane
# [match][sigh] add support for Apple Silicon Macs in iOS/iPadOS provisioning profiles

### Checklist

### Motivation and Context


Resolves #18581

Macs with Apple Silicon (AS) are capable of running native iOS/iPadOS apps. During development this only works if the Mac is added to the provisioning profile (`development` or `ad-hoc`).

### Description


This PR introduces a new device-class (`APPLE_SILICON_MAC`) that is used for iOS/iPadOS provisioning profiles to include AS Mac hardware. Including these devices is required to natively run apps on AS during development or using an ad-hoc signing. 

To include AS Macs, a new parameter has been added to match (and sigh): `include_mac_in_profiles`. Default is `false`. If enabled the provisioning profile device count and newly generated profiles will include registered AS Macs. 

If no AS Mac is registered in the Developer Portal, this setting won't have any effect. If run again without the `include_mac_in_profiles` flag, it will regenerate the provisioning profile without the AS Macs. (So it makes sense to include this setting in the `Matchfile` to keep it consistent)

### Testing Steps


1. Make sure to have at least one Apple Silicon Mac registered in the Developer Portal.
2. Create a provisioning profile with `fastlane match development --include_mac_in_profiles`
3. Check in Developer Portal that the newly created profile contains the AS Mac
4. Rerunning `fastlane match` without `include_mac_in_profiles` set, will regenerate the profile without the AS Macs.
